crates/starknet/src/transactions.rs

```diff
@@ -211,6 +210,43 @@ impl StarknetTransaction {
             _ => Ok(TransactionReceipt::Common(common_receipt)),
         }
     }
+
+    pub fn get_l2_to_l1_messages(&self) -> Vec<MsgToL1> {
```

Comment:
This looks like it could resolve https://github.com/0xSpaceShard/starknet-devnet-rs/issues/236
If so, can you edit your PR description to link it as closeable?

Reply:
I mean, it could resolve the issue if this line is edited: https://github.com/0xSpaceShard/starknet-devnet-rs/blob/72c11f63aa01a6b02a55d09cdcd3cdfe5c58a1cc/crates/types/src/rpc/transactions.rs#L123

I don't think this conversation should be marked as resolved until that is done (to rely on GitHub's unresolved conversation merge blocker). Let me know if I simply fail to see the responsible commit in the commit history. Also, let's try to supply the commit link to a conversation before closing, e.g. as done here: https://github.com/0xSpaceShard/starknet-devnet-rs/pull/235#discussion_r1377651364

Reply:
Oh sorry, totally missed it. You're right I read to fast.
For sure, I'll do.

Reply:
Will add testing in the next commits to considered this as a candidate for resolution. :+1:

Reply:
After merging #233, this PR now has a minor conflict with the base branch. Just replace the TODO comment that I introduced with the code you added to address this conversation.

Reply:
Included, thank you @FabijanC for the todo.